go/src/koding/klient/machine/machinegroup/ssh_test.go

```diff
@@ -0,0 +1,166 @@
+package machinegroup
+
+import (
+	"io/ioutil"
+	"os"
+	"os/user"
+	"strconv"
+	"testing"
+	"time"
+
+	"koding/kites/tunnelproxy/discover/discovertest"
+	"koding/klient/machine"
+	"koding/klient/machine/client/clienttest"
+)
+
+func TestSSH(t *testing.T) {
+	var (
+		builder = clienttest.NewBuilder(nil)
+		id      = machine.ID("serv")
+	)
+
+	wd, err := ioutil.TempDir("", "ssh")
+	if err != nil {
+		t.Fatalf("want err = nil; got %v", err)
+	}
+	defer os.RemoveAll(wd)
+
+	g, err := New(testOptions(wd, builder))
+	if err != nil {
+		t.Fatalf("want err = nil; got %v", err)
+	}
+	defer g.Close()
+
+	// Create test machine with IP address.
+	const fakeHost = "127.0.32.123"
+	createReq := &CreateRequest{
+		Addresses: map[machine.ID][]machine.Addr{
+			id: {
+				clienttest.TurnOnAddr(),
+				machine.Addr{
+					Network:   "ip",
+					Value:     fakeHost,
+					UpdatedAt: time.Now(),
+				},
+			},
+		},
+	}
+	if _, err := g.Create(createReq); err != nil {
```

Comment:
first to note: i didnt follow the whole flow. Would creating this with fake host twice cause problems? ie db unique index issues etc